Code/IO/include/sitkImageViewer.h

```diff
@@ -0,0 +1,205 @@
+  static const std::vector<std::string> & GetGlobalDefaultSearchPath();
+  /**@}*/
+
+  /** \brief Set/Get name list used to find the viewing application
+   * @{
+   */
+  static void SetGlobalDefaultExecutableNames( const std::vector<std::string> & names );
+  static const std::vector<std::string> & GetGlobalDefaultExecutableNames();
+  /**@}*/
+
+  /** \brief Set/Get the full path to the viewing application.
+   *
+   * The SetApplication method expects a full path name.  Using
+   * this method overrides the default application search.
+   * @{
+   */
+  void SetApplication( const std::string & app );
+  const std::string & GetApplication() const;
+  /**@}*/
+
+  /** \brief Set/Get command string used to launch the viewing application
+   *
+   * This command string may include the following %tokens:
+   *
+   * \li \c '%a' for the image viewing application (Fiji by default)
+   * \li \c '%f' for SimpleITK's temporary image file
+   *
+   *  For example, the default command string on Linux systems is:
+   *
+   *  \code
+   *  %a -o %f
+   *  \endcode
+   *
+   *  After token substitution it may become:
+   *
+   *  \code
+   *  /opt/Fiji.app/ImageJ-linux64 -o /tmp/Temp-65535-0.mha
+   *  \endcode
+   *
+   *  For another example, the default command string on Mac OS X is:
+   *
+   *  \code
+   *  open -a %a -n --args -eval \'open(\"%f\"); rename(\"%t\"); \'
+   *  \endcode
+   *
+   *  After token substitution the string may become:
+   *
+   *  \code
+   *  open -a /Applications/Fiji.app -n --args -eval 'open("/tmp/TempFile-29696-0.mha"); rename("/tmp/TempFile-29696-0.mha");'
+   *  \endcode
+   *
+   *  The string after \c '-eval' is an ImageJ macro the opens the file and sets the title of the window.
+   *
+   *  If the \c '%f' token is not found in the command string, the temporary file name is automatically
+   *  appended to the command argument list.
+   *
+   *  Note: Using the ImageViewer::SetCommand method overrides the default command and/or the
+   *  SITK_SHOW_COMMAND environment variable.
+   * @{
+   */
+  void SetCommand( const std::string & command );
+  const std::string & GetCommand() const;
+  /**@}*/
+
+  /** \brief Set/Get file extension of the temporary image file (default='.nii')
+   *
+   * If the viewing application is Fiji, the default image format is [MetaIO](https://itk.org/Wiki/ITK/MetaIO), '.mha'.
+   *
+   * If the viewing application is ImaegJ, the default image format is [NifTi](https://nifti.nimh.nih.gov/nifti-1/), '.nii'.  This is because by default ImageJ does not come with the MetaIO plugin.
+   * @{
+   */
+  void SetFileExtension( const std::string & ext );
+  const std::string & GetFileExtension() const;
+  /**@}*/
+
+  /** \brief Set/Get Debug flag.
+   *
+   * Turning on the Debug flag produces output showing the command string used to
+   * launch the viewing application.
+   * @{
+   */
+  static void SetGlobalDebug( const bool dbg );
+  static bool GetGlobalDebug();
```

Comment:
Is this a Global Default? or is there only one global state?

Please add On/Off methods as well.